### ISSUE TITLE
Make cypari2 ready for pari > 2.13

### ISSUE THREAD

**Issue by kliem** [sagemath/cypari2#111]
https://github.com/sagemath/cypari2/runs/4385764056?check_suite_focus=true

    Testing cypari2.convert
    ================================================================================
    Testing cypari2.gen
    ***   Warning: undefined environment variable: SHELL.
    **********************************************************************
    File "/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/cypari2/gen.cpython-36m-x86_64-linux-gnu.so", line ?, in cypari2.gen.Gen.__complex__
    Failed example:
    complex(g)
    Differences (ndiff with -expected +actual):
    - (2.2847006554165614+0j)
    + (1.118033988749895+0j)
    ***   Warning: undefined environment variable: SHELL.
    **********************************************************************
    File "/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/cypari2/gen.cpython-36m-x86_64-linux-gnu.so", line ?, in cypari2.gen.__test__.Gen.__complex__ (line 2006)
    Failed example:
    complex(g)
    Differences (ndiff with -expected +actual):
    - (2.2847006554165614+0j)
    + (1.118033988749895+0j)
    **********************************************************************
    2 items had failures:
    1 of  13 in cypari2.gen.Gen.__complex__
    1 of  13 in cypari2.gen.__test__.Gen.__complex__ (line 2006)
    ***Test Failed*** 2 failures.
    ================================================================================
    Testing cypari2.handle_error
    ***   user warning: test
    ================================================================================
    Testing cypari2.pari_instance
    /opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/doctest.py:1330: RuntimeWarning: cypari2 leaked 146652904 bytes on the PARI stack
    compileflags, 1), test.globs)

**Comment by videlec** [comment]
The offending test is
```
        >>> g = pari.quadgen(5) + pari('2/3')
        >>> complex(g)
        (2.2847006554165614+0j)
```
Which does make sense as `sqrt(5) / 2 + 7 / 6 ~ 2.28470065541656`. The answer you obtain is actually a floating point approximation of `sqrt(5) / 2 ~ 1.11803398874989`... it looks like 7/6 part of the expression was forgotten.

**Comment by videlec** [comment]
Only `2.2847006554165614+0j` (the old answer) is correct.

**Comment by videlec** [comment]
I am trying to figure out where this comes from. Seems to be the call to `gtofp`. Not sure who is wrong pari or cypari...

**Comment by videlec** [comment]
Note that the following are also all wrong
```
>>> float(pari.quadgen(5) + pari('2/3'))
1.118033988749895
>>> pari.quadgen(5) + pari('2/3') + pari('0.0')
1.11803398874989
```
But this one is correct
```
>>> pari('0.0') + pari.quadgen(5) + pari('2/3')
2.28470065541656
```

**Comment by videlec** [comment]
This is reproducible in GP I will fill a bug report
```
? quadgen(5) + 2/3 + 0.0
%1 = 1.1180339887498948482045868343656381177
? 0.0 + quadgen(5) + 2/3
%2 = 2.2847006554165615148712535010323047844
```

**Comment by kliem** [comment]
So our cypari workflow actually detected an upstream defect. That's a bit sad, especially since we detected it before the release (but I didn't really know what the failure was all about).

**Comment by videlec** [comment]
PARI upstream report https://pari.math.u-bordeaux.fr/cgi-bin/bugreport.cgi?bug=2419

**Comment by videlec** [comment]
The bug in PARI is now fixed in master. There is nothing we can do but wait for another release.

**Comment by mkoeppe** [comment]
But shouldn't have a new release so that we make progress on the Python 3.11 port?

**Comment by mkoeppe** [comment]
> So our cypari workflow actually detected an upstream defect. That's a bit sad, especially since we detected it before the release (but I didn't really know what the failure was all about).

Indeed. There's a clear lack of integration testing on the pari side. Too bad that they don't have github repo, or I could contribute my CI (https://trac.sagemath.org/ticket/33338)

**Comment by kliem** [comment]
I think @videlec meant that there is nothing that we can do, but wait for a release for pari. I guess we can just move forward with the next release, so that cypari works with python 3.11.

**Comment by mkoeppe** [comment]
Yes, let's do that

**Comment by tornaria** [comment]
For easy reference, the fix to pari-2.15.0 is the following:

https://pari.math.u-bordeaux.fr/cgi-bin/gitweb.cgi?p=pari.git;a=commit;h=9c4bcd3f497f30f757c8eecadc9b45b33754a540

Just  remove the first hunk patching the file CHANGES since it will not apply otherwise.

**Comment by videlec** [comment]
fixed by #124
